models.py

```diff
@@ -316,12 +357,29 @@ def _compute_emissions(self,
             # shape: (batch_size, seq_length, total_suffix_emb_size)
             affix_features = embedded_suffixes
 
-        # shape: (batch_size, seq_length, (2*window + 1) * word_embedding_size)
-        inputs = word_features
         if affix_features is not None:
             # shape: (batch_size, seq_length, (2*window+1)*word_embedding_size+total_affix_emb_size)
             inputs = torch.cat((inputs, affix_features), dim=-1)
 
+        # apply Bidirectional LSTM
+        if self.uses_lstm:
+            seq_lengths = torch.sum(mask.int(),dim = 1)
+            seq_lengths, sent_perm = seq_lengths.sort(0, descending=True)
+            inputs = inputs[sent_perm]
+
+            packed_input = pack_padded_sequence(inputs, seq_lengths.cpu().data.numpy(), batch_first=True)
```

Comment:
Can't this work without moving `seq_lengths` to CPU nor converting it to numpy array? e.g. just `seq_lengths.data`.